### PR TITLE
Added charts to album detail page

### PR DESCRIPTION
**Pull request by localastronaut**
Needs spacing / indenting work

### CODE REVIEW

File: templates/music/album_detail.html [hunk 1]
```diff
@@ -12,58 +13,142 @@
...
+            ]
+
+        }
+
+
+    </script>
+        <br/>
+        <br/>
+        <h4>Streams / Downloads</h4>
+{% for disc in album.discs %}
```
Comment (RevolutionTech) [inline templates/music/album_detail.html]: This section really doesn't make any sense at all. It looks like you're copying the whole script several times, once for every track in the album. This most definitely does not seem like what you're trying to do here.

File: templates/music/album_detail.html [hunk 2]
```diff
@@ -12,58 +13,142 @@
...
+    {% for track in disc %}
+    <head>
```
Comment (RevolutionTech) [inline templates/music/album_detail.html]: You definitely shouldn't have a `<head>` section here.

File: templates/music/album_detail.html [hunk 3]
```diff
@@ -12,58 +13,142 @@
...
+        <title>Bar Chart</title>
+        <script src="../Chart.js"></script>
+    </head>
+    <body>
```
Comment (RevolutionTech) [inline templates/music/album_detail.html]: This part of the HTML is already in the `<body>` tag, so you don't need the body tag defined here.

File: templates/music/album_detail.html [hunk 4]
```diff
@@ -12,58 +13,142 @@
...
+        <div class="bar-chart">
+                <canvas id="bar-canvas" height="450" width="600"></canvas>
+        </div>
+        <script>
```
Comment (RevolutionTech) [inline templates/music/album_detail.html]: Same JavaScript notes as before. This should be part of the `{% block extrajs %}` block and should have the type defined.

File: templates/music/album_detail.html [hunk 5]
```diff
@@ -12,58 +12,131 @@
...
+    <h3>REVENUE</h3>
+        <div class="bar-chart">
+            <canvas id="line-canvas" height="650px" width="1000px"></canvas>
+        </div>
+    </div>
+
+    <br/>
+    <br/>
+    <h3>Streams / Downloads</h3>
+    {% for disc in album.discs %}
```
Comment (RevolutionTech) [inline templates/music/album_detail.html]: The `{% for %}` loops don't make sense here. I think what you want to do is have one bar chart, but repeat the track names across the chart, right?

You should just remove these `{% for %}` and `{% endfor %}` tags. Also don't forget to remove to replace the label `{{ track.name }}` with "Track 1". I can apply a commit on this branch later and do the looping over the track names.

File: templates/music/album_detail.html [hunk 6]
```diff
@@ -12,58 +12,131 @@
...
+    {% for track in disc %}
+        <div class="bar-chart">
```
Comment (RevolutionTech) [inline templates/music/album_detail.html]: Once you remove the for loops, you'll want to unindent the bar chart.

File: templates/music/album_detail.html [hunk 7]
```diff
@@ -12,58 +12,131 @@
...
+                <canvas id="bar-canvas" height="450" width="600"></canvas>
+        </div>
+        {% block extrajs %}
```
Comment (RevolutionTech) [inline templates/music/album_detail.html]: The entire `{% block extrajs %}` and all JS should come after the current content block that we're in.

File: templates/music/album_detail.html [hunk 8]
```diff
@@ -12,58 +12,131 @@
...
+                },
+                {
+                    fillColor : "rgba(151,187,205,0.5)",
+                    strokeColor : "rgba(151,187,205,0.8)",
+                    highlightFill : "rgba(151,187,205,0.75)",
+                    highlightStroke : "rgba(151,187,205,1)",
+                    data : [randomScalingFactor(),randomScalingFactor(),randomScalingFactor(),randomScalingFactor(),randomScalingFactor(),randomScalingFactor(),randomScalingFactor()]
+                }
+            ]
+
```
Comment (RevolutionTech) [inline templates/music/album_detail.html]: nit: Remove empty lines

File: templates/music/album_detail.html [hunk 9]
```diff
@@ -12,58 +12,131 @@
...
+        window.myLine = new Chart(ctx).Line(lineChartData, {
+            responsive: true
+        });
+            var ctx = document.getElementById("bar-canvas").getContext("2d");
+            window.myBar = new Chart(ctx).Bar(barChartData, {
+                responsive : true
+            });
+
+    }
+
```
Comment (RevolutionTech) [inline templates/music/album_detail.html]: nit: Remove empty lines

File: templates/music/album_detail.html [hunk 10]
```diff
@@ -12,58 +12,131 @@
...
+                    strokeColor : "rgba(151,187,205,0.8)",
+                    highlightFill : "rgba(151,187,205,0.75)",
+                    highlightStroke : "rgba(151,187,205,1)",
+                    data : [randomScalingFactor(),randomScalingFactor(),randomScalingFactor(),randomScalingFactor(),randomScalingFactor(),randomScalingFactor(),randomScalingFactor()]
+                }
+            ]
+
+
+        }
+        window.onload = function(){
```
Comment (RevolutionTech) [inline templates/music/album_detail.html]: nit: Add a newline above `window.onload`

File: templates/music/album_detail.html [hunk 11]
```diff
@@ -12,58 +12,131 @@
...
+        var ctx = document.getElementById("line-canvas").getContext("2d");
+        window.myLine = new Chart(ctx).Line(lineChartData, {
+            responsive: true
+        });
+            var ctx = document.getElementById("bar-canvas").getContext("2d");
```
Comment (RevolutionTech) [inline templates/music/album_detail.html]: Unindent this section. Also, notice that you'll redefining `ctx`? You should use two different variable names, perhaps `line_context` and `bar_context` so that you don't override the previous.

File: templates/music/album_detail.html [hunk 12]
```diff
@@ -12,58 +12,131 @@
...
+    }
+
+
+        </script>
+        {% endblock %}
+
+        {% endfor %}
+    {% endfor %}
+
+    <div class="break-small">
```
Comment (RevolutionTech) [inline templates/music/album_detail.html]: As stated above, this HTML should come before the JavaScript.

File: templates/music/album_detail.html [hunk 13]
```diff
@@ -12,58 +12,131 @@
...
+        <hr>
+    </div>
+    <script type="text/javascript" src="{% static "js/Chart.js" %}"></script>
```
Comment (RevolutionTech) [inline templates/music/album_detail.html]: This script definition should be part of the `{% block extrajs %}` block.

File: templates/music/album_detail.html (hunk 11 above)
Comment (localastronaut) [inline templates/music/album_detail.html]: Which part do I need to rename? Not totally which part I need to re-name

File: templates/music/album_detail.html (hunk 5 above)
Comment (localastronaut) [inline templates/music/album_detail.html]: So do I just need to have "Track 1" instead of {{ track.name }}?

Comment (RevolutionTech) [inline templates/music/album_detail.html]: Yep, exactly.

File: templates/music/album_detail.html (hunk 11 above)
Comment (RevolutionTech) [inline templates/music/album_detail.html]: Notice how you have two lines that start out as `var ctx`? Those lines assign the context of a graph to the variable `ctx`, but the second definition is overriding the second one.

What you mean to do is have two _different_ variables, that are not related. Change the first line to
```
var line_context = document.getElement...
```
and the second line to
```
var bar_context = document.getElement...
```

**Comment by RevolutionTech** [comment]
It looks good. I can make the rest of the changes.
